vpr/src/route/rr_graph_clock.h

```diff
@@ -0,0 +1,119 @@
+#ifndef RR_GRAPH_CLOCK_H
+#define RR_GRAPH_CLOCK_H
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <unordered_map>
+#include <set>
+#include <utility>
+
+#include "clock_fwd.h"
+
+#include "clock_network_types.h"
+#include "clock_connection_types.h"
+
+class ClockNetwork;
+class ClockConnection;
+
+class SwitchPoint {
+    public:
+        // [grid_width][grid_height][nodes]
+        std::vector<std::vector<std::vector<int>>> rr_node_indices;
+        std::set<std::pair<int, int>> locations; // x,y
+    public:
+        /** Getters **/
+        std::vector<int> get_rr_node_indices_at_location(int x, int y) const;
+
+        std::set<std::pair<int, int>> get_switch_locations() const;
+
+        /** Setters **/
+        void insert_node_idx(int x, int y, int node_idx);
+};
+
+class SwitchPoints {
+    public:
+        std::unordered_map<std::string, SwitchPoint> switch_name_to_switch_location;
+    public:
+        /** Getters **/
+        std::vector<int> get_rr_node_indices_at_location(
+            std::string switch_name,
+            int x,
+            int y) const;
+
+        std::set<std::pair<int, int>> get_switch_locations(std::string switch_name) const;
+
+        /** Setters **/
+        void insert_switch_node_idx(std::string switch_name, int x, int y, int node_idx);
+};
+
```

Comment:
Consider renaming to ClockRRGraphBuilder

Reply:
Done: commit 2788ef1706edf3991263d1845e9420d42b983573